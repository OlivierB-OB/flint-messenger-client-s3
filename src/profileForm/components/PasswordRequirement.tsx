import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import CheckCircleOutline from '@material-ui/icons/CheckCircleOutline';
import RadioButtonUnchecked from '@material-ui/icons/RadioButtonUnchecked';
import React from 'react';

export interface IPasswordRequirementProps {
  check: boolean;
  message: string;
}

export function PasswordRequirement({ check, message }: IPasswordRequirementProps) {
  return (
    <Grid container item alignItems="center" style={{ fontSize: 'small' }}>
      {check ? (
        <CheckCircleOutline style={{ fontSize: 'inherit' }} />
      ) : (
        <RadioButtonUnchecked style={{ fontSize: 'inherit' }} />
      )}
      <Typography style={{ fontSize: 'inherit' }}>{message}</Typography>
    </Grid>
  );
}
