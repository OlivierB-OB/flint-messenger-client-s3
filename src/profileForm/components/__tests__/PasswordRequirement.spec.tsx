import React from 'react';
import { shallow } from 'enzyme';
import { PasswordRequirement } from '../PasswordRequirement';

describe('PasswordRequirement', () => {
  it('should display requirement', async () => {
    const component = shallow(<PasswordRequirement check={true} message="description" />);
    expect(component).toMatchSnapshot();
  });

  it('should not be checked if check is false', async () => {
    const component = shallow(<PasswordRequirement check={false} message="description" />);
    expect(component).toMatchSnapshot();
  });
});
