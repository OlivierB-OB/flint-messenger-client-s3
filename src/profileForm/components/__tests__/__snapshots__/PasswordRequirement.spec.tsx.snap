// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordRequirement should display requirement 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  item={true}
  style={
    Object {
      "fontSize": "small",
    }
  }
>
  <CheckCircleOutlineIcon
    style={
      Object {
        "fontSize": "inherit",
      }
    }
  />
  <WithStyles(ForwardRef(Typography))
    style={
      Object {
        "fontSize": "inherit",
      }
    }
  >
    description
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`PasswordRequirement should not be checked if check is false 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  item={true}
  style={
    Object {
      "fontSize": "small",
    }
  }
>
  <RadioButtonUncheckedIcon
    style={
      Object {
        "fontSize": "inherit",
      }
    }
  />
  <WithStyles(ForwardRef(Typography))
    style={
      Object {
        "fontSize": "inherit",
      }
    }
  >
    description
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Grid))>
`;
